application/forms/ChannelForm.php

```diff
@@ -110,14 +114,29 @@ protected function assemble()
         );
 
         if ($this->channelId !== null) {
+            $isInUse = RuleEscalationRecipient::on($this->db)
+                    ->columns('1')
+                    ->filter(Filter::all(
+                        Filter::equal('deleted', 'n'),
+                        Filter::any(
+                            Filter::equal('channel_id', $this->channelId),
+                            Filter::equal('contact.default_channel_id', $this->channelId)
+                        )
+                    ))
+                    ->count();
+
             /** @var FormSubmitElement $deleteButton */
             $deleteButton = $this->createElement(
                 'submit',
                 'delete',
                 [
                     'label'          => $this->translate('Delete'),
                     'class'          => 'btn-remove',
-                    'formnovalidate' => true
+                    'formnovalidate' => true,
+                    'disabled'       => $isInUse,
+                    'title'          => $isInUse
+                        ? $this->translate('Cannot delete channel, contacts or event rules are using it')
```

Comment:
*Channel is still referenced as a contact's default channel or in an event rule's escalation*